### PR TITLE
Add reusable zinc functions from scaffoldmaker

### PR DESCRIPTION
**Pull request by rchristie**
Consistently use name argument instead of field_name.
Some minor bug fixes.
Fix names of field nodeset mean and range functions as not limited to coordinates.

### CODE REVIEW

File: src/opencmiss/utils/zinc/field.py [hunk 1]
```diff
@@ -411,6 +412,114 @@ def get_or_create_field_coordinates(fieldmodule : Fieldmodule, name="coordinates
...
+    :return: Zinc FieldGroup.
+    """
+    field = fieldmodule.findFieldByName(name)
+    if field.isValid():
+        group = field.castGroup()
+        if group:
+            return group
+    return create_field_group(fieldmodule, name, managed=True)
+
+def group_get_or_create_nodeset_group(group : FieldGroup, nodeset : Nodeset) -> NodesetGroup:
```
Comment (hsorby) [inline src/opencmiss/utils/zinc/field.py]: I think this should be `get_or_create_group_nodeset_group` of `get_or_create_nodeset_group`.

Comment (rchristie) [reply src/opencmiss/utils/zinc/field.py]: Decided it's better to create the FieldNodeGroup instead; caller can call .getNodesetGroup() on it. Note FieldNodeGroup can be created either in or out of a group but most common use is likely to be in the group.

File: src/opencmiss/utils/zinc/finiteelement.py [hunk 2]
```diff
@@ -172,41 +172,41 @@ def get_node_name_centres(nodeset: Nodeset, coordinatesField: Field, nameField:
...
+    with ChangeManager(fieldmodule):
+        mean_field = fieldmodule.createFieldNodesetMean(field, nodeset)
         fieldcache = fieldmodule.createFieldcache()
-        result, meanCoordinates = meanCoordinatesField.evaluateReal(fieldcache, componentsCount)
+        result, mean_values = mean_field.evaluateReal(fieldcache, components_count)
         assert result == RESULT_OK
-        del meanCoordinatesField
+        del mean_field
         del fieldcache
     assert result == RESULT_OK
```
Comment (hsorby) [inline src/opencmiss/utils/zinc/finiteelement.py]: I don't think this second assert is required.

File: src/opencmiss/utils/zinc/field.py [hunk 3]
```diff
@@ -399,24 +397,159 @@ def create_field_coordinates(fieldmodule : Fieldmodule, name="coordinates", comp
...
     """
     Get or create a stored mesh location field for storing locations in the
     supplied mesh, used for storing data projections.
     Note can't currently verify existing field stores locations in the supplied mesh.
-    Asserts that existing field of name is a Stored Mesh Location field.
+    New field is managed by default.
+
     :param fieldmodule:  Zinc fieldmodule to find or create field in.
     :param mesh:  Mesh to store locations in, from same fieldmodule.
     :param name:  Name of new field. If None, defaults to "location_" + mesh.getName().
```
Comment (hsorby) [inline src/opencmiss/utils/zinc/field.py]: Tempted to change this to `:param name:  Name of new field. If not defined defaults to "location_" + mesh.getName().`

File: src/opencmiss/utils/zinc/field.py [hunk 4]
```diff
@@ -399,24 +397,159 @@ def create_field_coordinates(fieldmodule : Fieldmodule, name="coordinates", comp
...
+
+def create_field_stored_mesh_location(fieldmodule : Fieldmodule, mesh : Mesh, name=None, managed=False) -> FieldStoredMeshLocation:
+    """
+    Create a stored mesh location field for storing locations in the
+    supplied mesh, used for storing data projections.
+    New field is not managed by default.
+
+    :param fieldmodule:  Zinc fieldmodule to find or create field in.
+    :param mesh:  Mesh to store locations in, from same fieldmodule.
+    :param name:  Name of new field. If None, defaults to "location_" + mesh.getName().
```
Comment (hsorby) [inline src/opencmiss/utils/zinc/field.py]: As below applies here also.

File: src/opencmiss/utils/zinc/finiteelement.py (hunk 2 above)
Comment (rchristie) [reply src/opencmiss/utils/zinc/finiteelement.py]: Fixed

File: src/opencmiss/utils/zinc/field.py (hunk 4 above)
Comment (rchristie) [reply src/opencmiss/utils/zinc/field.py]: Fixed.

File: src/opencmiss/utils/zinc/field.py (hunk 3 above)
Comment (rchristie) [reply src/opencmiss/utils/zinc/field.py]: Fixed.

File: src/opencmiss/utils/zinc/finiteelement.py (hunk 2 above)
Comment (rchristie) [reply src/opencmiss/utils/zinc/finiteelement.py]: Fixed.

**Pull request merged**